### ISSUE TITLE
Remove slow decorator cont.

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#3869]
## 🧰 Task
Following on from #3705 it would be nice to remove our remaining uses of `@pytest.mark.timeout` as it continues to cause spurious test fails - see discussion in https://github.com/napari/napari/pull/3734#issuecomment-999786044.

The remaining uses of `@pytest.mark.timeout` cover cases where we want to make sure that computation is not triggered on a full dask array for things like calculating our contrast limits - see https://github.com/napari/napari/blob/3069f2dcc94a00c9ae4643fd171325ac2afb05dd/napari/layers/utils/_tests/test_layer_utils.py#L76-L83 for example.

We added this test as we were previously having lots of slowdowns at this point.

I can see a couple different options.

1. We could just drop the decorator. Tests will keep passing. If we accidentally trigger compute on full array then we'll get massive arrays generated and test suite will fail ungracefully.
2. We could drop the test entirely, leaving regressions untested. (But then hopefully enable more benchmarking with asv where we'd see a temporal regression).
3. Rewrite the test so that it doesn't have the timeout but will fail more gracefully by potentially letting us check for compute using smaller arrays that won't crash the system. We might be able to use something modified off of https://github.com/pydata/xarray/blob/5e8de55321171f95ed9684c33aa47112bb2519ac/xarray/tests/test_dask.py#L39-L43 and https://github.com/pydata/xarray/blob/5e8de55321171f95ed9684c33aa47112bb2519ac/xarray/tests/__init__.py#L127-L132 from xarray.

At this point I'm open to any of those three options. My least favorite is (1) as it seems like just kicking a problem down the road. My most favourite is probably (3), but that might take some work.

I can't imagine we're the first project to try and write a test like this either though, so we might be able to ask the dask / xarray folks for help.

What do @GenevieveBuckley @tlambert03 @jni and others think?

**Comment by jni** [comment]
Yeah, I like option 3 best. @tlambert03 had a pattern somewhere that counted the number of actual chunk computations in a dask array, which can probably be reused here? I think I wrote this test (?) a while back before I'd seen that pattern.

**Comment by brisvag** [comment]
Dumb dask newbie idea: could we use `delayed` to wrap a function that bumps a `counter` of some sort, then make a dask array out of that? This way we can just look at the counter to see how many times it was called.

**Comment by tlambert03** [comment]
not dumb!  that's pretty much exactly the pattern that @jni is referring to.  we use it here:

https://github.com/napari/napari/blob/b88926c1fe14d2a938d059d8cf8d4da6043a324f/napari/layers/_tests/test_dask_layers.py#L18-L30

(we can't just use `@delayed` because we want it to look like an array, so we use `map_blocks`)

**Comment by tlambert03** [comment]
actually, just realized there's a fixture below that you can probably reuse:

https://github.com/napari/napari/blob/b88926c1fe14d2a938d059d8cf8d4da6043a324f/napari/layers/_tests/test_dask_layers.py#L88

(that one does use delayed along with `da.stack` to make it like an array)

**Comment by psobolewskiPhD** [comment]
Greping, I dont find `pytest.mark.timeout` in the code base.
Maybe we can close this?
